app/code/community/Algolia/Algoliasearch/Model/Source/MaxRecordSize.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+class Algolia_Algoliasearch_Model_Source_MaxRecordSize
+{
+    protected $_options;
+
+    public function toOptionArray()
+    {
+        if (!$this->_options) {
+            $options[] = Algolia_Algoliasearch_Helper_Config::DEFAULT_MAX_RECORD_SIZE;
```

Comment:
Does this mean that 10kb will be always the default value and will need to be changed manually by the customer?
If so - how about changing the form input to  "disabled" `text` and set it's value always based on what is coming from internal API? If internal API fails - use 10kb.

Reply:
WDYT?

Reply:
I rewrote the getMaxRecordSize() to fetch and save config if config does not have a value but cache the default value if for instance the API key isn't set. I remove the field as well since this is not changeable.